core-bundle/src/Resources/contao/dca/tl_search_term.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+$GLOBALS['TL_DCA']['tl_search_term'] = array
+(
+	// Config
+	'config' => array
+	(
+		'sql' => array
+		(
+			'keys' => array
+			(
+				'id' => 'primary',
+				'term' => 'unique'
```

Comment:
```suggestion
				'term' => 'unique',
				'documentFrequency' => 'index'
```

This fixes the DELETE deadlocks for me: `An exception occurred while executing 'DELETE FROM tl_search_term WHERE documentFrequency = 0':  SQLSTATE[40001]: Serialization failure: 1213 Deadlock found when trying to get lock; try restarting transaction`